pkg/e2e/operators/operators.go

```diff
@@ -125,6 +127,131 @@ func checkSecrets(h *helper.H, namespace string, secrets []string) {
 	})
 }
 
+func getInstallPlan(h *helper.H, sub *operatorv1.Subscription) (*operatorv1.InstallPlan, error) {
+	subNamespace := sub.Namespace
+	subName := sub.Name
+	err := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+		s, err := h.Operator().OperatorsV1alpha1().Subscriptions(subNamespace).Get(subName, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		if s.Status.InstallPlanRef != nil {
+			sub = s
+			return true, nil
+		}
+
+		return false, nil
+	});
+	if err != nil {
+		return nil, err
+	}
+
+	return h.Operator().OperatorsV1alpha1().InstallPlans(subNamespace).Get(sub.Status.InstallPlanRef.Name, metav1.GetOptions{})
+}
+
+func approveInstallPlan(h *helper.H, ip *operatorv1.InstallPlan) error {
+	ip.Spec.Approved = true
+	_, err := h.Operator().OperatorsV1alpha1().InstallPlans(ip.Namespace).Update(ip)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ensureCSVIsInstalled(h *helper.H, csvName string, namespace string) error {
+	err := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+		csv, err := h.Operator().OperatorsV1alpha1().ClusterServiceVersions(namespace).Get(csvName, metav1.GetOptions{})
+		if err != nil && !kerror.IsNotFound(err) {
+			return false, err
+		}
+
+		if csv.Status.Phase == operatorv1.CSVPhaseSucceeded {
+			return true, nil
+		}
+		return false, nil
+	});
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getChannel() string {
```

Comment:
I'm not sure if this is the correct way to do this. The channel name is different on different environments.

Reply:
Looks good to me. 

Do you know if there are other uses of this function? It might make more sense to put this logic in something like pkg/common or pkg/osd depending on how widely used it would be.

If you don't know any off the top of your head we'll keep it here and address it later.

Reply:
This is specific to the CatalogSource resource so I think it's ok to leave this here.
fyi a small update -> https://github.com/openshift/osde2e/pull/242/commits/4ed4cf675da0c4ffec38db6342658f7f47eb9b0a